textattack/constraints/grammaticality/cola.py

```diff
@@ -0,0 +1,79 @@
+import lru
+import nltk
+from transformers import AutoModelForSequenceClassification, AutoTokenizer
+
+from textattack.commands.attack.attack_args import HUGGINGFACE_DATASET_BY_MODEL
+from textattack.constraints import Constraint
+from textattack.models.wrappers import HuggingFaceModelWrapper
+
+
+class COLA(Constraint):
+    """Constrains an attack to text that has a similar number of linguistically
+    accecptable sentences as the original text. Linguistic acceptability is
+    determined by a model pre-trained on the `CoLA dataset <https://nyu-
+    mll.github.io/CoLA/>`_. By default a BERT model is used, see the `pre-
+    trained models README.
+
+    <https://github.com/QData/TextAttack/tree/master/textattack/models>`_ for a
+    full list of available models.
+
+    Args:
+        max_diff (float): The absolute maximum difference between the percent of valid
+            sentences in the reference text and the percent of valid sentences in the
+            attacked text
+        model_name (str): The name of the pre-trained model to use for classification.
+        compare_against_original (bool): If `True`, compare against the original text.
+            Otherwise, compare against the most recent text.
+    """
+
+    def __init__(
+        self,
+        max_diff,
+        model_name="bert-base-uncased-cola",
+        compare_against_original=True,
+    ):
+        super().__init__(compare_against_original)
+        if not isinstance(max_diff, float):
+            raise TypeError("max_diff must be a float")
+        if max_diff < 0.0 or max_diff > 1.0:
+            raise ValueError("max_diff must be a value between 0.0 and 1.0")
+        if model_name not in HUGGINGFACE_DATASET_BY_MODEL:
+            raise ValueError(
+                "model_name must be the name of a pre-trained model on the CoLA dataset"
```

Comment:
One small suggestion for error message is that we might want to tell the user what they supplied for `model_name`. So like `f"model_name {model_name} must be ....`

Reply:
Good catch, but by allowing users to supply their own model, I don't think we'll be able to have a check like this anymore.

Reply:
Yes.